providers/curation/github.js

```diff
@@ -139,11 +140,42 @@ class GitHubCurationService {
       fileBody.committer = { name: info.name || info.login, email: info.email }
     if (get(currentContent, '_origin.sha')) {
       fileBody.sha = currentContent._origin.sha
+
+      try {
+        let validateResult = await this._validateComponentDefinitionExists(patch)
+        for (var elem of validateResult) {
+          if (elem.revision) {
+            return response.status(400).send({ errors: `component does not exist: ${elem.revision.toString()}` })
+          }
+        }
+      } catch (error) {
+        return response.status(400).send({ errors: `error in validating patch ${error.toString()}` })
+      }
       return serviceGithub.repos.updateFile(fileBody)
     }
     return serviceGithub.repos.createFile(fileBody)
   }
 
+  // If a definition exists then return the patch.
+  // Otherwise return an array of just the coordinates of the definitions that do not exist
+  async _validateComponentDefinitionExists(patch) {
+    const result = await Promise.all(
+      map(patch.revisions, async (_, key) => {
```

Comment:
should use `throat` (perhaps 10) as there may be many revisions

Reply:
agreed.